### ISSUE TITLE
NO_FCP running in Gitlab serving React SPA with "serve" module

### ISSUE THREAD

**Issue by dericgw** [GoogleChrome/lighthouse-ci#196]
I am attempting to automate Lighthouse into our Gitlab pipeline and I cannot get it to run a successful test using the [ziet/serve](https://github.com/zeit/serve) module. Locally, this works fine, but when running in our pipeline, I get NO_FCP error. What is interesting is that when I serve the project using Webpack Dev Server to serve the site, it works as expected. Of course, the metrics are horrible because it is a dev server. 

Here is the section of my `gitlab-ci.yml`:

```
perf_test:
  image: cypress/browsers:node10.16.0-chrome77
  dependencies:
    - build
  stage: test
  script:
    - npm install
    - npm run lighthouse // lhci autorun
  artifacts:
    paths:
      - ./report.html
    expire_in: 7 days
```

And here is my `lighthouserc.json`

```
{
  "ci": {
    "collect": {
      "numberOfRuns": 3,
      "settings": {
        "chromeFlags": "--no-sandbox --headless",
        "useThrottling": true,
        "disable-storage-reset": true,
        "onlyCategories": [
          "performance",
          "accessibility",
          "best-practices"
        ]
      },
      "startServerCommand": "npm run serve",
      "url": [
        "http://localhost:6001/create-account"
      ]
    },
    "upload": {
      "target": "temporary-public-storage"
    }
  }
}

```

Any idea why this would not be running in Gitlab but runs fine locally and also why it works in Gitlab when I serve the site from Webpack Dev Server?

**Comment by patrickhulce** [comment]
Thanks for filing! That sounds strange indeed. Are you just using `serve` to serve static files and have you tried using the lhci built-in static server instead?

It'd look instead like

```json5
{
  "ci": {
    "collect": {
      "staticDistDir": "./dist", // assuming it's default webpack, use your real output directory if different
      "url": ["http://localhost/create-account"]
    }
  }
}

```

Other debugging ideas... maybe you can curl your server `/create-account` in Gitlab and make sure the output is what you expect?

Also just FYI, it should already be `--headless` by default so you shouldn't need to pass that flag.

**Comment by dericgw** [comment]
@patrickhulce Thanks for such a quick reply! I did not know that I could use the built in server without having to define a port number (and I thought those were random, ephemeral ports), so it looks like I was making it harder on myself than it had to be. I will try that now.

**EDIT**
I found the issue. I am setting the PUBLIC_URL environment variable when I build the front-end assets and it points to an S3 bucket. That is why Lighthouse was not able to find my site and that is why I was getting the NO_FCP (No First Contentful Paint).

Thanks for your quick reply and your help with this.
